[MDEP-689] Fixes ignored dependency filtering in go-offline goal

ldhardy opened the pull request:
Duplicate issues [MDEP-725](https://issues.apache.org/jira/browse/MDEP-725) and [MDEP-568](https://issues.apache.org/jira/browse/MDEP-568)

```
$ mvn -B dependency:go-offline  -DexcludeGroupIds=org.dojotoolkit -DexcludeClassifiers=distribution -DexcludeArtifactIds=org.dojotoolkit:dojo:zip:1.10.8,dojo -DexcludeTypes=zip
```
Results in attempted resolution of all of the excluded parameters.  Looking at the GoOfflineMojo class, the filters were clearly never wired into the doExecute logic.  This PR fixes that and adds some unit and integration tests.



Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEP) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MDEP-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MDEP-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

slawekjaranowski reviewed (approved):
We are working on get rid of m-artifact-trafsfer - MDEP-858
but change looks ok.

slawekjaranowski commented on src/main/java/org/apache/maven/plugins/dependency/resolvers/GoOfflineMojo.java:
```diff
@@ -152,6 +169,46 @@ protected Set<Artifact> resolvePluginArtifacts() throws DependencyResolverExcept
         return resolveDependableCoordinate(buildingRequest, dependableCoordinates, "plugins");
     }
 
+    protected FilterArtifacts getArtifactsFilter() {
+        final FilterArtifacts filter = new FilterArtifacts();
+
```
We have the same method in `ResolvePluginsMojo` - we can move it to `AbstractDependencyFilterMojo`

slawekjaranowski reviewed (changes requested):
duplicate code

ldhardy commented:
Do you know what is required to get a Jira account so I can make comments on the tickets?  I've tried emailing various aliases and done the self-serve account request with no luck. It's been almost 2 months since my request.  

I'll work on addressing your comments here in the next week or so.

> We are working on get rid of m-artifact-trafsfer

I'm not sure how this relates to this PR, can you give me more info about what you mean?

mthmulders commented:
>  I've tried emailing various aliases and done the self-serve account request with no luck.

Apologies, that one must've slipped through. I approved your self-serve account request a minute ago.

slawekjaranowski commented on src/test/resources/unit/go-offline-test/exclude-plugin-config.xml:
```diff
@@ -0,0 +1,35 @@
+      </plugin>
+    </plugins>
+  </build>
+</project>
```
missing new line ... and in other files

slawekjaranowski commented:
> Do you know what is required to get a Jira account so I can make comments on the tickets? I've tried emailing various aliases and done the self-serve account request with no luck. It's been almost 2 months since my request.
> 
> I'll work on addressing your comments here in the next week or so.
> 
> > We are working on get rid of m-artifact-trafsfer
> 
> I'm not sure how this relates to this PR, can you give me more info about what you mean?

get rid of `m-artifact-transfer` it is not addressed by this PR
simply we add new code which use `m-artifact-transfer` .. but it is accepted for me now ... only please try to eliminate duplicate codes

elharo commented:
@slawekjaranowski Can we go ahead and merge this one?

jira-importer commented:
Resolve #1199

mschaefers commented:
any idea when this fix will be released?
